File: Regressions/ICT-21770/test/ict21770i/MiddleCCImpl.py
import ict21770__POA

from ict21770i.ParentCCImpl import ParentCCImpl
from Acspy.Util.BaciHelper import addProperty

class MiddleCCImpl(ict21770__POA.MiddleCC, ParentCCImpl):
    def __init__(self):
        ParentCCImpl.__init__(self)

    def initialize(self):
        ParentCCImpl.initialize(self)
        addProperty(self, "prop2")

    def printValue2(self):
        print(self._get_prop2().get_sync()[0])

    def printValues(self):
        self.printValue1()
        self.printValue2()

File: Regressions/ICT-21770/test/alma/ict21770/MiddleCCImpl/MiddleCCImpl.java
package alma.ict21770.MiddleCCImpl;

import alma.ACS.ROdouble;
import alma.ACS.ROdoubleHelper;
import alma.ACS.ROdoublePOATie;
import alma.ACS.impl.ROdoubleImpl;
import alma.ACSErr.CompletionHolder;
import alma.acs.container.ContainerServices;
import alma.acs.component.ComponentLifecycleException;

import alma.ACS.jbaci.DataAccess;
import alma.ACS.jbaci.MemoryDataAccess;

import alma.ict21770.MiddleCCOperations;
import alma.ict21770.ParentCCImpl.ParentCCImpl;

public class MiddleCCImpl extends ParentCCImpl implements MiddleCCOperations
{
    protected ROdouble prop2;

	public void initialize(ContainerServices containerServices) throws ComponentLifecycleException {
        super.initialize(containerServices);

        try {
            DataAccess prop2DA = new MemoryDataAccess();
            ROdoubleImpl prop2Impl = new ROdoubleImpl("prop2", this, prop2DA);
            ROdoublePOATie prop2Tie = new ROdoublePOATie(prop2Impl);
            prop2 = ROdoubleHelper.narrow(this.registerProperty(prop2Impl, prop2Tie));
        } catch (Throwable th) {
        	throw new ComponentLifecycleException("Failed to create properties.", th); 
        }
	}
    
	public void printValue2() {
        CompletionHolder ch = new CompletionHolder();
        System.out.println(prop2().get_sync(ch));
	}

	public void printValues() {
        printValue1();
        printValue2();
	}

	public ROdouble prop2() {
		return prop2;
	}
}
